<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ConvertTab</class>
 <widget class="QWidget" name="ConvertTab">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>700</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="spacing">
    <number>15</number>
   </property>
   <property name="leftMargin">
    <number>20</number>
   </property>
   <property name="topMargin">
    <number>20</number>
   </property>
   <property name="rightMargin">
    <number>20</number>
   </property>
   <property name="bottomMargin">
    <number>20</number>
   </property>
   <item>
    <widget class="QLabel" name="lblPipelineTitle">
     <property name="text">
      <string>Model Conversion Pipeline</string>
     </property>
     <property name="styleSheet">
      <string notr="true">font-size: 18px; font-weight: bold; color: #000000; padding: 5px;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="framePipeline">
     <property name="minimumHeight">
      <number>60</number>
     </property>
     <layout class="QHBoxLayout" name="pipelineLayout">
      <item>
       <spacer name="spacerLeft">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="lblPt">
        <property name="text">
         <string>  .pt (PyTorch)  </string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #4a148c; color: white; padding: 10px 20px; border-radius: 5px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblArrow1">
        <property name="text">
         <string>  ➜  </string>
        </property>
        <property name="styleSheet">
         <string notr="true">font-size: 20px; color: #666666;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblOnnx">
        <property name="text">
         <string>  .onnx (ONNX)  </string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #1565c0; color: white; padding: 10px 20px; border-radius: 5px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblArrow2">
        <property name="text">
         <string>  ➜  </string>
        </property>
        <property name="styleSheet">
         <string notr="true">font-size: 20px; color: #666666;</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="lblHef">
        <property name="text">
         <string>  .hef (Hailo)  </string>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: #e65100; color: white; padding: 10px 20px; border-radius: 5px; font-weight: bold;</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="spacerRight">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="mainContentLayout">
     <property name="spacing">
      <number>15</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="leftPanelLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupStep1">
         <property name="title">
          <string>Step 1: PyTorch → ONNX</string>
         </property>
         <layout class="QVBoxLayout" name="step1Layout">
          <property name="spacing">
           <number>12</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="ptFileLayout">
            <item>
             <widget class="QLabel" name="lblPtFile">
              <property name="text">
               <string>PT File:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editPtPath">
              <property name="placeholderText">
               <string>Select PyTorch model file (.pt)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnBrowsePt">
              <property name="text">
               <string>...</string>
              </property>
              <property name="browse" stdset="0">
               <string>true</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QFrame" name="frameModelDetection">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="styleSheet">
             <string notr="true">QFrame#frameModelDetection { background-color: #f5f5f5; border: 1px solid #e0e0e0; border-radius: 4px; padding: 6px; }</string>
            </property>
            <layout class="QHBoxLayout" name="modelDetectionLayout">
             <property name="spacing">
              <number>15</number>
             </property>
             <property name="topMargin">
              <number>4</number>
             </property>
             <property name="bottomMargin">
              <number>4</number>
             </property>
             <item>
              <widget class="QLabel" name="lblDetectionIcon">
               <property name="text">
                <string>Model:</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; color: #333333;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblYoloVersion">
               <property name="text">
                <string>-</string>
               </property>
               <property name="toolTip">
                <string>Detected YOLO version</string>
               </property>
               <property name="styleSheet">
                <string notr="true">color: #1565c0; font-weight: bold;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblSepTask">
               <property name="text">
                <string>|</string>
               </property>
               <property name="styleSheet">
                <string notr="true">color: #bdbdbd;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblTaskLabel">
               <property name="text">
                <string>Task:</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; color: #333333;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblTaskType">
               <property name="text">
                <string>-</string>
               </property>
               <property name="toolTip">
                <string>Detected task type (detect/segment/classify)</string>
               </property>
               <property name="styleSheet">
                <string notr="true">color: #2e7d32; font-weight: bold;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblSepOpset">
               <property name="text">
                <string>|</string>
               </property>
               <property name="styleSheet">
                <string notr="true">color: #bdbdbd;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblOpsetLabel">
               <property name="text">
                <string>Opset:</string>
               </property>
               <property name="styleSheet">
                <string notr="true">font-weight: bold; color: #333333;</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="lblRecommendedOpset">
               <property name="text">
                <string>-</string>
               </property>
               <property name="toolTip">
                <string>Recommended ONNX opset version</string>
               </property>
               <property name="styleSheet">
                <string notr="true">color: #ef6c00; font-weight: bold;</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="detectionSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </spacer>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="inputSizeLayout">
            <item>
             <widget class="QLabel" name="lblInputSize">
              <property name="text">
               <string>Input Size:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinWidth">
              <property name="minimum">
               <number>32</number>
              </property>
              <property name="maximum">
               <number>4096</number>
              </property>
              <property name="value">
               <number>640</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lblX">
              <property name="text">
               <string> x </string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinHeight">
              <property name="minimum">
               <number>32</number>
              </property>
              <property name="maximum">
               <number>4096</number>
              </property>
              <property name="value">
               <number>640</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lblBatch">
              <property name="text">
               <string>Batch:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinBatch">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>32</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="sizespacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="opsetLayout">
            <item>
             <widget class="QLabel" name="lblOpset">
              <property name="text">
               <string>Opset Ver:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboOpset">
              <item>
               <property name="text">
                <string>11</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>12</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>13</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>14</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>17</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <spacer name="opsetSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="onnxOutLayout">
            <item>
             <widget class="QLabel" name="lblOnnxOut">
              <property name="text">
               <string>Output:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editOnnxOutPath">
              <property name="placeholderText">
               <string>Output ONNX file path</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnBrowseOnnxOut">
              <property name="text">
               <string>...</string>
              </property>
              <property name="browse" stdset="0">
               <string>true</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="convertBtnLayout">
            <item>
             <spacer name="btnSpacer1">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnConvertOnnx">
              <property name="text">
               <string>Convert to ONNX</string>
              </property>
              <property name="minimumHeight">
               <number>40</number>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: #2e7d32; font-size: 14px; padding: 12px 30px;</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="btnSpacer2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupStep2">
         <property name="title">
          <string>Step 2: ONNX → HEF (Hailo Compile)</string>
         </property>
         <layout class="QVBoxLayout" name="step2Layout">
          <property name="spacing">
           <number>12</number>
          </property>
          <item>
           <layout class="QHBoxLayout" name="onnxFileLayout">
            <item>
             <widget class="QLabel" name="lblOnnxFile">
              <property name="text">
               <string>ONNX File:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editOnnxPath">
              <property name="placeholderText">
               <string>Select ONNX model file (.onnx)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnBrowseOnnx">
              <property name="text">
               <string>...</string>
              </property>
              <property name="browse" stdset="0">
               <string>true</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="calibLayout">
            <item>
             <widget class="QLabel" name="lblCalibDir">
              <property name="text">
               <string>Calib Dir:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editCalibPath">
              <property name="placeholderText">
               <string>Select calibration images directory</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnBrowseCalib">
              <property name="text">
               <string>...</string>
              </property>
              <property name="browse" stdset="0">
               <string>true</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="targetLayout">
            <item>
             <widget class="QLabel" name="lblTarget">
              <property name="text">
               <string>Target:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboTarget">
              <item>
               <property name="text">
                <string>hailo8</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>hailo8l</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>hailo15h</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="lblPerf">
              <property name="text">
               <string>Optimize:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboPerf">
              <item>
               <property name="text">
                <string>fps</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>accuracy</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>power</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <spacer name="targetSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="hefOutLayout">
            <item>
             <widget class="QLabel" name="lblHefOut">
              <property name="text">
               <string>Output:</string>
              </property>
              <property name="minimumWidth">
               <number>80</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="editHefOutPath">
              <property name="placeholderText">
               <string>Output HEF file path</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnBrowseHefOut">
              <property name="text">
               <string>...</string>
              </property>
              <property name="browse" stdset="0">
               <string>true</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="compileBtnLayout">
            <item>
             <spacer name="compileSpacer1">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnCompileHef">
              <property name="text">
               <string>Compile to HEF</string>
              </property>
              <property name="minimumHeight">
               <number>40</number>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: #ef6c00; font-size: 14px; padding: 12px 30px;</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="compileSpacer2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupLog">
       <property name="title">
        <string>Conversion Log</string>
       </property>
       <property name="minimumWidth">
        <number>400</number>
       </property>
       <layout class="QVBoxLayout" name="logLayout">
        <item>
         <widget class="QTextEdit" name="txtLog">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="html">
           <string>&lt;p style="color: #666666;"&gt;Ready for conversion...&lt;/p&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressConvert">
          <property name="value">
           <number>0</number>
          </property>
          <property name="minimumHeight">
           <number>25</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnClearLog">
          <property name="text">
           <string>Clear Log</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
